backend/src/layer/data/mail/mail_sender.rs

```diff
@@ -0,0 +1,171 @@
+    ///
+    /// # Errors
+    /// Returns an error, if the connection could not be established to the smtp server
+    pub fn new(config: MailInfo) -> Result<Self> {
+        let creds = Credentials::new(config.username.clone(), config.password.clone());
+        if let Ok(transport_builder) = SmtpTransport::relay(&config.smtp_server) {
+            let mailer = transport_builder
+                .port(config.smtp_port)
+                .credentials(creds)
+                .build();
+            Ok(Self { config, mailer })
+        } else {
+            Err(DataError::NoSuchItem)
+        }
+    }
+
+    fn try_notify_admin_image_report(&self, info: &ImageReportInfo) -> MailResult<()> {
+        let sender = self.get_sender()?;
+        let reciever = self.get_reciever()?;
+        let report = Self::get_report(info)?;
+        let email = Message::builder()
+            .from(sender)
+            .to(reciever)
+            .subject("An image was reported for reviewing")
+            .body(report)?;
+        self.mailer.send(&email)?;
+        info!("Email sent successfully!");
+        Ok(())
+    }
+
+    fn get_sender(&self) -> MailResult<Mailbox> {
+        format!("app <{}>", self.config.username.clone())
+            .parse()
+            .map_err(MailError::AddressError)
+    }
+
+    fn get_reciever(&self) -> MailResult<Mailbox> {
+        format!("admin <{}>", self.config.admin_email_address)
+            .parse()
+            .map_err(MailError::AddressError)
+    }
+
+    fn get_report(info: &ImageReportInfo) -> MailResult<String> {
+        let template_file_contents =
+            fs::read_to_string(REPORT_TEMPLATE_FILE).map_err(MailError::TemplateError)?;
+        let template = Template::new(&template_file_contents);
+        let mut args = HashMap::new();
+        let image_link: &str = &info.image_link;
+        args.insert("image_link", image_link);
+        let image_id: &str = &Simple::from_uuid(info.image_id).to_string();
+        args.insert("image_id", image_id);
+        let report_count: &str = &info.report_count.to_string();
+        args.insert("report_count", report_count);
+        let reason: &str = &info.reason.to_string();
+        args.insert("reason", reason);
+        let image_got_hidden: &str = &info.image_got_hidden.to_string();
+        args.insert("image_got_hidden", image_got_hidden);
+        let positive_rating_count: &str = &info.positive_rating_count.to_string();
+        args.insert("positive_rating_count", positive_rating_count);
+        let negative_rating_count: &str = &info.negative_rating_count.to_string();
+        args.insert("negative_rating_count", negative_rating_count);
+        let get_image_rank: &str = &info.get_image_rank.to_string();
+        args.insert("get_image_rank", get_image_rank);
+
+        Ok(template.render(&args))
+    }
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
consider adding a test to check that the values could be inserted into the template correctrly (and there are no wrongly named fields and so on; on whatever the library errors)